src/main/java/uk/gov/hmcts/dts/fact/model/Court.java

```diff
@@ -75,7 +72,8 @@ public Court(uk.gov.hmcts.dts.fact.entity.Court courtEntity) {
         this.crownLocationCode = courtEntity.getNumber();
         this.countyLocationCode = courtEntity.getCciCode();
         this.magistratesLocationCode = courtEntity.getMagistrateCode();
-        this.areasOfLaw = courtEntity.getAreasOfLaw().stream().map(AreaOfLaw::new).collect(toList());
+        this.areasOfLaw = this.stripUrlencodedFromAreaOfLaw(
+            courtEntity.getAreasOfLaw().stream().map(AreaOfLaw::new).collect(toList()));
```

Comment:
I don't think I understand why you're using map when you are mutating the original object but I guess it doesn't matter.

Reply:
Creating areasOfLaw can be imporved by adding something like ".map(this::stripUrlencodedFromAreaOfLaw)" to the stream processing, it will then only do a single loop over the list

Reply:
its because we mapping to an areaof law object, as thats what we need, and then i remove the urlencoded stuff.